File: matlab_test_files/system_example_matlab.py

import numpy as np
import scipy
import matcompat

# if available import pylab (from matlibplot)
try:
    import matplotlib.pylab as plt
except ImportError:
    pass

syst_fake = tf(np.array(np.hstack((1.))), np.array(np.hstack((1., 2., 3.))))
syst_fake_dis = c2d(syst_fake, 0.01)
[output, t] = plt.step(syst_fake_dis)
plt.plot(output)
out_len = length(output)
input = np.arange(1., (out_len)+1)
input[:] = 1.
[num, den] = stmcb_test(output, input, 0., 2.)
#% sys_model=tf(num,den,0.01)
#% step(sys_model)
#% hold on
#% step(syst_fake)

File: matlab_test_files/system_example_matlab.m
syst_fake=tf([1],[1 2 3])
syst_fake_dis=c2d(syst_fake,0.01)
[output,t]=step(syst_fake_dis)
plot(output)

out_len = length(output)
input=1:out_len;
input(:)=1;
[num,den]=stmcb_test(output,input,0,2);
sys_model=tf(num,den,0.01)
step(sys_model)
hold on
step(syst_fake)


% [b,a] = butter(6,0.2);              % Butterworth filter design
% h = filter(b,a,[1 zeros(1,100)]);   % Filter data using above filter
% [bb,aa] = stmcb_test(h,4,4)
